py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/s3_utils.py

```diff
@@ -0,0 +1,38 @@
+import boto3
+import os
+
+from collections.abc import Iterable
+from pathlib import Path
+
+def file_list_from_s3(bucket_name: str='mbta-gtfs-s3',
+                      file_prefix: str=None) -> Iterable[(str, int)]:
+    """
+    generate filename, filesize tuples for every file in an s3 bucket
+
+    :param bucket_name: the name of the bucket to look inside of
+    :param file_prefix: prefix for files to generate
+
+    :yield filename, filesize tuples from inside of the bucket
+    """
+    s3 = boto3.resource('s3')
+    bucket = s3.Bucket(bucket_name)
+
+    for bucket_object in bucket.objects.filter(Prefix=file_prefix):
+        yield (bucket_object.key, bucket_object.size)
```

Comment:
i guess we'll have to figure out a better way to do this.

the main s3 bucket has _so many files_.